src/ui/libs/DatalensChartkit/ChartKit/plugins/BlankChart/renderer/BlankChartWidget.tsx

```diff
@@ -35,19 +39,62 @@ const BlankChartWidget = (props: BlankChartWidgetProps) => {
         }
     }, []);
 
+    const chartState = React.useRef<any>({});
+
     React.useEffect(() => {
-        if (dimensions && originalData?.render) {
-            const content = originalData.render({...dimensions});
-            if (contentRef.current) {
-                contentRef.current.innerHTML = String(content ?? '');
-
-                const widgetRendering = Performance.getDuration(generatedId);
-                if (onLoad && widgetRendering) {
-                    onLoad({widget: props.data, widgetRendering});
+        chartStorage.set(generatedId, {
+            chartId: generatedId,
+            getState: () => chartState.current,
+            setState: (value: any, options?: {silent: boolean}) => {
+                chartState.current = merge({}, chartState.current, value);
+
+                if (!options?.silent) {
+                    render();
                 }
+            },
+        });
+
+        return () => {
+            chartStorage.delete(generatedId);
+        };
+    }, [generatedId, dimensions]);
```

Comment:
Do we actually need `dimensions` here?

Reply:
Yes, we need to get the current state when changing the dimensions